Develop

samuelouteda opened the pull request:
Assembled Home section:
![image](https://user-images.githubusercontent.com/74317636/101215624-8adfaf00-3654-11eb-8af4-6fd325a84e3f.png)

Fullname component and AnimatedRole sub-component:
![image](https://user-images.githubusercontent.com/74317636/101215871-ead65580-3654-11eb-9aa2-78ed9c2059b6.png)

BackgroundVid component:
![image](https://user-images.githubusercontent.com/74317636/101216002-22dd9880-3655-11eb-8ea3-a08d5a05436a.png)

Sidebar and Home sections rendered at the same time:
![image](https://user-images.githubusercontent.com/74317636/101216112-54eefa80-3655-11eb-810f-15d46b46efce.png)

samuelouteda committed:
parent 174d1b0a5d265e377cd8500471288ef9a70cdf64
author Samuel Outeda <samuelouteda@gmail.com> 1605795684 -0400
committer Samuel Outeda <samuelouteda@gmail.com> 1607723078 -0400

Creation of ProfilePic

Creation of the first component of sidebar, the ProfilePic, which contains a profile picture and a name

Creation of NavLinks

Creation of the second component of sidebar, the NavLinks, which contains a list of navigation links

Creating file to import all sections

Creating file to import all sections and export where needed

Simplifying imports

Using exports from index.js to simplify imports

Improving code

Reducing the amount of imports, converting functions into ES6 arrow function aways, adding missing end of line

Creation of Footer

Creation of the third and last component of sidebar, the Footer, which contains links to social networks with their icons and copyright statement.

Revert "Improving code"

This reverts commit 4bb030ab747fe710dcc43af68e501e6287ef8f0d.

Improving code

Reducing the amount of imports, converting functions into ES6 arrow function always, adding missing end of line, removing unnecessary border style.

Reducing the amount of imports, converting functions into ES6 arrow function always, adding missing end of line, removing unnecessary border style.

Assembling SideBar Component

Assembling SideBar Component by putting together its three child components: ProfilePic, NavLinks and Footer.

Adding EOFL

Adding EOFL

Adding EOFL

Adding EOFL

Improving code

Reducing the amount of imports, converting functions into ES6 arrow function always, adding missing end of line, removing unnecessary border style, converting if...else statement into ternary operator.

Correcting typo error

Correcting an uppercase letter that should be lowercase

Refactoring list

Refactoring list to be the return of a function, avoiding code repeat

Adding letter to Avatar

Adding one more letter between Avatar tags to use a better acronym

Refactoring list

Refactoring list to use less callback functions

Refactoring links

Refactoring links to be the return of a function, avoiding code repeat

Adding EOFL

Adding EOFL

Removing border style

Removing unnecessary border style.

Creation of style rules

Creation of style rules for sizes and colors

Adding style rules

Adding more thickness rules

Changing style rules

Changing style rules to use predefined rules

Adding semi-colon

Adding missing semi-colon

Changing style rules

Changing style rules to use predefined rules

Correcting typo error

Correcting wrong name of variable

Adding import

Adding missing import of style rules

Adding style rules

Adding thickness style rule

Changing style rules

Changing style rules to use predefined rules

Adding style rules

Adding thickness style rule

Changing style rules

Changing thickness style rule

Changing style rules

Changing style rules to use predefined rules

Adding style rules

Adding style rules and a new classification for thickness values, with a comment explaining why

Changing style rules

Changing style rules to use new value

Creation of IconMaker

Creation of a fourth component of sidebar, the IconMaker, which contains logic to return diferent icons based on received props.

Refactoring Footer

Refactoring the component Footer to use the new component IconMaker

Refactoring NavLinks

Refactoring the component NavLinks to use the new component IconMaker

Refactoring imports

Refactoring the Icon Maker component to use less imports

Refactoring imports

Refactoring the component to use less imports

Changing import

Importing without specifying /index

Moving export

Moving export to the left

Adding and editing style rules

Adding thickness rules and editing color rules

Changing style rules

Changing styles  to use new rules

Changing style rules

Changing styles  to use new rules

Changing style rules and passing props

Changing styles to use new rules and passing props to ProfilePic

Changing style rules and receiving props

Changing styles to use new rules and receiving props from SideBar.jsx to determine the picture displayed

Adding base project

Revert "Improving code"

This reverts commit 4bb030ab747fe710dcc43af68e501e6287ef8f0d.

Reducing the amount of imports, converting functions into ES6 arrow function always, adding missing end of line, removing unnecessary border style.

Adding EOFL

Adding EOFL

Adding EOFL

Adding EOFL

Correcting typo error

Correcting an uppercase letter that should be lowercase

Adding EOFL

Adding EOFL

Correcting typo error

Correcting wrong name of variable

Creation of IconMaker

Creation of a fourth component of sidebar, the IconMaker, which contains logic to return diferent icons based on received props.

Refactoring Footer

Refactoring the component Footer to use the new component IconMaker

Refactoring NavLinks

Refactoring the component NavLinks to use the new component IconMaker

Refactoring imports

Refactoring the component to use less imports

Changing import

Importing without specifying /index

Moving export

Moving export to the left

Adding and editing style rules

Adding thickness rules and editing color rules

Changing style rules

Changing styles  to use new rules

Changing style rules

Changing styles  to use new rules

Changing style rules and passing props

Changing styles to use new rules and passing props to ProfilePic

Changing style rules and receiving props

Changing styles to use new rules and receiving props from SideBar.jsx to determine the picture displayed

Adding dependencies

Adding @material-ui/core and @material-ui/icons dependencies to the project

Assembling SideBar Component

Assembling SideBar Component by putting together its three child components: ProfilePic, NavLinks and Footer.
Adding EOFL

Adding EOFL
Adding EOFL

Adding EOFL
Improving code

Reducing the amount of imports, converting functions into ES6 arrow function always, adding missing end of line, removing unnecessary border style, converting if...else statement into ternary operator.
Correcting typo error

Correcting an uppercase letter that should be lowercase
Refactoring list

Refactoring list to be the return of a function, avoiding code repeat
Adding letter to Avatar

Adding one more letter between Avatar tags to use a better acronym
Refactoring list

Refactoring list to use less callback functions
Refactoring links

Refactoring links to be the return of a function, avoiding code repeat
Adding EOFL

Adding EOFL
Removing border style

Removing unnecessary border style.
Creation of style rules

Creation of style rules for sizes and colors
Adding style rules

Adding more thickness rules
Changing style rules

Changing style rules to use predefined rules
Adding semi-colon

Adding missing semi-colon
Changing style rules

Changing style rules to use predefined rules
Correcting typo error

Correcting wrong name of variable
Adding import

Adding missing import of style rules
Adding style rules

Adding thickness style rule
Changing style rules

Changing style rules to use predefined rules
Adding style rules

Adding thickness style rule
Changing style rules

Changing thickness style rule
Changing style rules

Changing style rules to use predefined rules
Adding style rules

Adding style rules and a new classification for thickness values, with a comment explaining why
Changing style rules

Changing style rules to use new value
Creation of IconMaker

Creation of a fourth component of sidebar, the IconMaker, which contains logic to return diferent icons based on received props.
Refactoring Footer

Refactoring the component Footer to use the new component IconMaker
Refactoring NavLinks

Refactoring the component NavLinks to use the new component IconMaker
Refactoring imports

Refactoring the component to use less imports
Changing import

Importing without specifying /index
Moving export

Moving export to the left
Adding and editing style rules

Adding thickness rules and editing color rules
Changing style rules

Changing styles  to use new rules
Changing style rules

Changing styles  to use new rules
Changing style rules and passing props

Changing styles to use new rules and passing props to ProfilePic
Changing style rules and receiving props

Changing styles to use new rules and receiving props from SideBar.jsx to determine the picture displayed

Moving generic components

Moving generic components to root level

Creation of index at component level

Creation of index at component level to import components and then export them, simplifying the imports in SideBar.jsx

Simplifying imports

Simplifying imports using new index.js

Changing name of folder

Changing name of folder, removing the ".js" at the end

Changing import

Change import of IconMaker, due to the new name of its main folder

Receiving props

Receiving props correctly

Refactoring imports

Refactoring named imports to be default individual imports, because the @material-ui/icons dependency does not allow named imports for that components

Correcting import

Correcting the relative path of an import

Editing render method

Editing render method to render SideBar component

Changing styles

Changing styles to be more responsive

Changing border style

Changing border style to be on one side only

Removing "/"

Removing "/" from import's relative path

Correcting typo error

Correcting typo error in the URL of a link

Improving security

Improving security of links

Creation of ProfilePic

Creation of the first component of sidebar, the ProfilePic, which contains a profile picture and a name

Creation of NavLinks

Creation of the second component of sidebar, the NavLinks, which contains a list of navigation links

samuelouteda committed:
Improving code

Adding Font Styles to index.html, importing the new object ICONS in IconMaker/index.js, improving definition of PropTypes in NavLinks, adding Font Styles to NavLinks, ProfilePic and Footer, changing the size of SideBar from '16%' to '200px', removing min and max width from all the components of sidebar, adding transition effects to the social icons of Footer, adding 'position: fixed' to SideBar, saving the links of the social media in a LINKS object on constants.js and importing it on sidebar to be used in Footer as props, moving constants.js to utils directory (there is helpers.js too), changing the style rule of width to 200px and adding style rules of transition durations

samuelouteda committed:
Improving code

Adding Font Styles to index.html, importing the new object ICONS in IconMaker/index.js, improving definition of PropTypes in NavLinks, adding Font Styles to NavLinks, ProfilePic and Footer, changing the size of SideBar from '16%' to '200px', removing min and max width from all the components of sidebar, adding transition effects to the social icons of Footer, adding 'position: fixed' to SideBar, saving the links of the social networks in a LINKS object on constants.js and importing it on sidebar to be used in Footer as props, moving constants.js to utils directory (there is helpers.js too), changing the style rule of width to 200px and adding style rules of transition durations

samuelouteda committed:
Merge remote-tracking branch 'origin/develop' into develop

samuelouteda committed:
Refactoring SideBar and applying some minor improvements

Refactoring SideBar to hide in extra-small sizes and render an AppBar, adding breakpoint to ProfilePic to hide its text in extra-small sizes, adding event listener to NavLinks via props and its corresponding PropTypes to hide the SideBar in extra-small sizes when clicking a link, adding more TransitionDuration rules to styles.js, updating Footer to use the new TransitionDuration rules

samuelouteda committed:
Improving Responsive Design

Importing IconMaker on SideBar.jsx to render the HomeIcon, increasing range of breakpoint in which "overflow: auto" is applied, changing the title of the Appbar from "Carlos Valero" to "My Portfolio", and its center method to use flexbox instead of calculating manually the necessary left spacing, positioning with "absolute" the MenuIcon, adding breakpoints to drawerPaper class,  increasing the range of Hidden elements to show the Appbar in small sizes, removing the breakpoint in ProfilePic to always show the title "Carlos Valero" below the image, refactoring transitionDuration rules in styles.js to be less complicated and more intuitive, using the new transitionDuration rules in Footer, importing the MenuIcon in IconMaker and adding logic to render it depending on received props, adding the necessary keyword "burger" in ICONS object to render the MenuIcon, importing MenuIcon in IconMaker/index.js and then exporting it to be imported in IconMaker

samuelouteda committed:
Creation of Home Section

Creation of Home Section assembling two components and one sub-component: BackgroundVid, which displays a background video (video_03_gaussian_blur) that fills the entire screen and behind of that renders a background image (image_1) in case the video fails to load; AnimatedRole, a sub-component which contains logic to simulate a typing animation in a string passed as props; and Fullname, which renders a main title with a Fade animation and displays the AnimatedRole sub-component underneath. The three components are imported in an index.js file and then exported to be used in Home.jsx. The App.js file has been modified to render the new Home.jsx component

samuelouteda committed:
Creation of Home Section

Creation of Home Section assembling two components and one sub-component: BackgroundVid, which displays a background video (video_03_gaussian_blur) that fills the entire screen and behind of that renders a background image (image_1) in case the video fails to load; AnimatedRole, a sub-component which contains logic to simulate a typing animation in a string passed as props; and Fullname, which renders a main title with a Fade animation and displays the AnimatedRole sub-component underneath. The three components are imported in an index.js file and then exported to be used in Home.jsx. The App.js file has been modified to render the new Home.jsx component

kilordpepo commented on public/index.html:
```diff
@@ -2,6 +2,10 @@
 <html lang="en">
   <head>
     <meta charset="utf-8" />
+    <link
```
Can you investigate if this can be imported differntly without using a link in the index.html?

samuelouteda replied:
Maybe using [Google's Webfontloader](https://github.com/typekit/webfontloader), with an <script> like this:
```
<script>
  WebFont.load({
    google: {
      families: ['Roboto Condensed', 'Merriweather']
    }
  });
</script>
```
Or importing them in a css file

kilordpepo replied:
Yeah, I think this one it's better!

kilordpepo commented on src/components/NavLinks/SideBarLinks.js:
```diff
@@ -16,11 +16,10 @@ import IconMaker from "../IconMaker/IconMaker";
     color: "gray",
-    textTransform: "uppercase"
+    textTransform: "uppercase",
+    fontFamily: "Roboto Condensed, sans-serif"
```
Add this font to the css rules we have and use it here

kilordpepo commented on src/components/NavLinks/SideBarLinks.js:
```diff
@@ -85,8 +89,9 @@ const SelectedListItem = ({ items, iconImage }) => {
 SelectedListItem.propTypes = {
-  items: PropTypes.arrayOf(PropTypes.array),
-  iconImage: PropTypes.arrayOf(PropTypes.string)
+  items: PropTypes.arrayOf(PropTypes.arrayOf(PropTypes.string)),
```
can we conver this to something more readable? I would think it's better to have an array of objects then just accest the different properties of the object insted of position 0 and 1

kilordpepo commented on src/components/ProfilePic/SideBarProfile.js:
```diff
@@ -6,19 +6,17 @@ import { styles } from "../../styles/styles";
   text: {
     color: "white",
     backgroundColor: styles.colors.transparentLightBlue,
+    fontFamily: "Merriweather, serif",
```
add all different font families to the css rule file we have

kilordpepo commented on src/sections/Home/Home.jsx:
```diff
@@ -0,0 +1,22 @@
+
+const useStyles = makeStyles(theme => ({
+  home: {
+    width: "100%",
```
I believe we have a style for 100% already?

kilordpepo commented on src/sections/Home/components/AnimatedRole/HomeRole.js:
```diff
@@ -0,0 +1,103 @@
+import React, { useState, useEffect } from "react";
+import PropTypes from "prop-types";
+import { makeStyles, Typography } from "@material-ui/core";
+import { styles } from "../../../../styles/styles";
```
I think you can start with "/" instead of "./" and this would make the url shorter

kilordpepo commented on src/sections/Home/components/AnimatedRole/HomeRole.js:
```diff
@@ -0,0 +1,103 @@
+const useStyles = makeStyles(theme => ({
+  text2: {
+    color: "white",
+    fontFamily: "Roboto Condensed, sans-serif",
```
same comment about fonts

kilordpepo commented on src/sections/Home/components/AnimatedRole/HomeRole.js:
```diff
@@ -0,0 +1,103 @@
+
+const useStyles = makeStyles(theme => ({
+  text2: {
+    color: "white",
```
use colors we have define

kilordpepo commented on src/sections/Home/components/AnimatedRole/HomeRole.js:
```diff
@@ -0,0 +1,103 @@
+    color: "white",
+    fontFamily: "Roboto Condensed, sans-serif",
+    [theme.breakpoints.only("xs")]: {
+      fontSize: "1.1rem"
```
I would also recommend define this font sizes in the rule file

kilordpepo commented on src/sections/Home/components/AnimatedRole/HomeRole.js:
```diff
@@ -0,0 +1,103 @@
+
+const AnimatedRole = ({ messages }) => {
+  const classes = useStyles();
+  const [state, setState] = useState({
```
Sepparate the state into different states asi this too generic and any change would cause the component to re-render. Only group variables into the same state if they are part of a group that makes sense

kilordpepo commented on src/sections/Home/components/BackgroundVid/HomeVid.js:
```diff
@@ -0,0 +1,55 @@
+import React from "react";
+//import { styles } from "../../../../styles/styles";
+import { makeStyles } from "@material-ui/core";
+import video from "./video/video_03_gaussian_blur.mp4";
```
can you investigate if this is the best video format we can have for the site?

kilordpepo commented on src/sections/Home/components/BackgroundVid/HomeVid.js:
```diff
@@ -0,0 +1,55 @@
+import React from "react";
+//import { styles } from "../../../../styles/styles";
```
remove comment?

kilordpepo commented on src/sections/Home/components/BackgroundVid/HomeVid.js:
```diff
@@ -0,0 +1,55 @@
+        loop='loop'
+      >
+        <source src={video} type='video/mp4' />
+        <source src='' type='video/webm' />
```
Is this ok?

kilordpepo commented on src/sections/Home/components/Fullname/HomeFullname.js:
```diff
@@ -0,0 +1,84 @@
+import AnimatedRole from "../AnimatedRole/HomeRole";
+
+const useStyles = makeStyles(theme => ({
+  homeCont: {
```
same comment about styles here and everywhere, I believe you can use alot of the rules you have defined already

kilordpepo commented on src/sections/SideBar/SideBar.jsx:
```diff
@@ -41,22 +97,63 @@ const SideBar = () => {
+            paper: classes.drawerPaper
+          }}
+          ModalProps={{
+            keepMounted: true // Better open performance on mobile.
```
remove comment

kilordpepo reviewed (changes requested):
Please make some changes as it seems there are still things we can optimize

samuelouteda committed:
Changing style rules and improving code

Adding color, fontFamilies, fontSizes and sizes rules to styles.js and applying them in ProfilePic, NavLinks, Footer, SideBar.jsx, BackgroundVid, Fullname and Home.jsx components; changing items props in SideBar to be an array of objects instead of an array of arrays, changing the items PropTypes in NavLinks to define the received object via PropTypes.shape, separating the states in AnimatedRole component into two: state and blink. State contains and object with related states that works together, and blink an individual state which determines when the cursor needs to start blinking. The state of 'typingSpeed' was removed and replaced by a variable called 'speed'. Adding new webm video format to BackgroundVid.

kilordpepo commented on src/sections/Home/components/BackgroundVid/HomeVid.js:
```diff
@@ -0,0 +1,56 @@
+      >
+        <source src={videoMp4} type='video/mp4' />
+        <source src={videoWebm} type='video/webm' />
+        <source src='' type='video/ogg' />
```
remove this

samuelouteda replied:
I finally found a way to convert the video! :smile: 
Should I delete it anyway? :sob:

kilordpepo replied:
hehe ok, leave it then 👍

kilordpepo commented on src/sections/SideBar/SideBar.jsx:
```diff
@@ -1,30 +1,86 @@
+    position: "fixed",
+    top: 0,
+    left: 0,
+    zIndex: "999",
```
change to number

kilordpepo replied:
define z-index rules

kilordpepo reviewed (commented):
Last row of comments

samuelouteda committed:
Adding zIndex rules and improving code

Adding zIndex rules to styles.js, using these rules in SideBar.jsx, Fullname and BackgroundVid, adding index.js in styles folder to export styles.js and then be imported where it is needed, updating imports of styles.js in SideBar.jsx, ProfilePic, NavLinks, Footer, AnimatedRole, Fullname, BackgroundVid and Home.jsx to use the new index.js, importing new video .ogv file in BackgroundVid and using it in a <source> element; updating the name of states from "state" to "animation", deleting "hide" state because "start" can fulfill the same function (hide was unnecessary) and moving UseEffect Hooks right before return statement in AnimatedRole, reducing imports on Home.jsx and SideBar.jsx deleting the unnecessary "/index" termination

samuelouteda committed:
Fixing errors in rebase and importing fonts with Google's Webfontloader

Fixing incorrect changes applied by the rebase (duplicated files), importing fontfamilies in index.html with Google's Webfontloader, additionally importing Roboto's fontfamily and adding it to styles.js and using the new rule for Roboto Condensed in NavLinks, Footer and AnimatedRole components

samuelouteda committed:
Merge remote-tracking branch 'origin/develop' into develop

samuelouteda committed:
Importing fonts with Google's Webfontloader

Importing Roboto, Roboto Condensed and Merriweather with Google's Webfontloader, creating rule in styles.js for Roboto and modifying rule for Roboto Condensed and using this new rule in NavLinks, Footer and AnimatedRole components.
